test/tests/meshgenerators/patterned_mesh_generator/tests

```diff
@@ -1,13 +1,36 @@
 [Tests]
-  [./patterned_mesh_generator_test]
+  design = 'meshgenerators/PatternedMeshGenerator.md'
+  issues = '#11640 #23312'
+
+  [patterned_mesh_generator_test]
     type = 'Exodiff'
     input = 'patterned_mesh_generator.i'
     exodiff = 'patterned_mesh_generator_out.e'
     mesh_mode = 'REPLICATED'
     recover = false
+    requirement = 'The system shall have the ability to stitch multiple input meshes together into a new mesh based upon a user-defined pattern.'
+  []
+
+  [boundary_rename_test]
+    type = 'Exodiff'
+    input = 'boundary_rename.i'
+    cli_args = "--mesh-only"
+    exodiff = 'boundary_rename_in.e'
+    mesh_mode = 'REPLICATED'
+    recover = false
+    requirement = 'The system shall have the ability to stitch multiple input meshes with differing boundary ids together into a new mesh based upon a user-defined pattern.'
+  []
+
+  [errors]
+    requirement = 'The system shall report a reasonable error when stitching meshes together from a user-defined pattern when'
 
-    requirement = 'The system shall have the ability to stetch multiple input meshes together into a new mesh based upon a user-defined pattern.'
-    design = 'meshgenerators/PatternedMeshGenerator.md'
-    issues = '#11640'
-  [../]
+    [non_existent_boundary]
+      type = 'RunException'
+      input = 'patterned_mesh_generator.i'
+      cli_args = 'Mesh/pmg/right_boundary="fake_boundary"'
+      expect_err = 'does not exist in input mesh'
+      detail = 'the boundary name does not exist'
+      mesh_mode = 'REPLICATED'
+    []
+  []
```

Comment:
add an error test for the inconsistent size of boundary_ids

Reply:
I am actually having a difficult time coming up with an input that triggers this error without hitting the error (here)[https://github.com/idaholab/moose/blob/74817397957818d058304850d3fca6d7ceecd352/framework/src/meshgenerators/PatternedMeshGenerator.C#L136-L145] first. Do you have any ideas? If not, I think the inconsistent size test can be removed.

Reply:
just use one with 4 sides and one with 1 side all around?

Reply:
In such a case, there is no correct way to set left/right/top/bottom_boundary. See below:

`[Mesh]
  [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 5
    ny = 5
    subdomain_ids = '1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     1 1 1 1 1
                     '
  []

  [gmg2]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 5
    ny = 5
    subdomain_ids = '2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     2 2 2 2 2
                     '
  []

  [all_sides_same]
    type = RenameBoundaryGenerator
    input = 'gmg2'
    old_boundary = 'left right top bottom'
    new_boundary = 'left left left left'
  []

  [pmg]
    type = PatternedMeshGenerator
    inputs = 'gmg all_sides_same'
    pattern = '1 0 ;
               0 1'
  []
[]`

The error:

`*** ERROR ***
(Mesh/pmg/inputs):
    The 'right_boundary' parameter with value 'right' does not exist in input mesh 'all_sides_same'`

Now if I set left/right/top/bottom_boundary all to 'left':

`  [pmg]
    type = PatternedMeshGenerator
    inputs = 'gmg all_sides_same'
    pattern = '1 0 ;
               0 1'
    left_boundary = left
    right_boundary = left
    top_boundary = left
    bottom_boundary = left
  []`

This runs because the boundary id for the 'left' mesh in each input mesh is a single number, thus the error is not triggered.